kyaml/yaml/types.go

```diff
@@ -48,28 +48,43 @@ func IsEmptyMap(node *RNode) bool {
 	if IsEmpty(node) {
 		return true
 	}
+	return IsYNodeEmptyMap(node.YNode())
+}
 
-	return node.YNode().Kind == yaml.MappingNode && len(node.YNode().Content) == 0
+// IsNil return true if the node is nil, or its underlying YNode is nil.
+func IsNil(node *RNode) bool {
+	return node == nil || node.YNode() == nil
 }
 
 func IsNull(node *RNode) bool {
```

Comment:
So we only change the internal implementation of these functions and keep the function signature unchanged?